coderd/prometheusmetrics/aggregator.go

```diff
@@ -0,0 +1,249 @@
+package prometheusmetrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/codersdk/agentsdk"
+)
+
+const (
+	// MetricHelpForAgent is a help string that replaces all agent metric help
+	// messages. This is because a registry cannot have conflicting
+	// help messages for the same metric in a "gather". If our coder agents are
+	// on different versions, this is a possible scenario.
+	metricHelpForAgent = "Metric is forwarded from workspace agent connected to this instance of coderd."
```

Comment:
```suggestion
	metricHelpForAgent = "Metrics are forwarded from workspace agents connected to this instance of coderd."
```

Suggestion: I think it's preferable to use the plural form here.

Side-note: In a HA setup with multiple coderd's, would metrics only be for those few agents connected to that specific coderd instance? Or is it global?

Reply:
> Suggestion: I think it's preferable to use the plural form here.

Fixed

> Side-note: In a HA setup with multiple coderd's, would metrics only be for those few agents connected to that specific coderd instance? Or is it global?

Only connected ones as metrics are not fetched from a database but just cached in memory. I think that it applies to most of Coder metrics, right?